design-proposals/nic-hotplug.md

```diff
@@ -0,0 +1,134 @@
+# Overview
+This KubeVirt design proposal discusses possible implementations for adding -
+and removing - network interfaces from running Virtual Machines, without
+requiring a restart.
+
+## Motivation
+Hot-plug / hot-unplug (add / remove) nics to running VMs is an industry
+standard available in multiple platforms.
+This would help bridge the gap between KubeVirt's feature set and what the user
+base expects from it.
+
+## Goals
+Adding network interfaces to running VMs.
+Removing networking interfaces from running VMs.
+A VM can have multiple interfaces connected to the same (secondary) network(s).
+
+## Non Goals
+Providing multiple connections to the primary cluster network (managed by
+Kubernetes).
+
+## Definition of Users
+This feature is intended for cluster users who want to connect their existing
+Virtual Machines to additional existing networks.
+These users may - or may not - have permissions to create the aforementioned
+additional networks; if so, they would need to rely on the cluster
+administrator to provision it for them.
+
+## User Stories
+* as a user, I want to add a new network interface to a running VM.
```

Comment:
user story should contain the motivation / "because", could you explain why is it useful and why users can't list all the needed networks while defining the VM?

Reply:
Doesn't that defeat the purpose of the motivation section ?

Reply:
You only argue there that it is an industry standard. I think it is important to rephrase why is hotplug needed / what is it used for, so reviewer of this proposal can decide whether the feature is something that makes sense in the scope of Kubernetes. But I guess you could mention it in Motivation instead of here

Reply:
Possibly "because my workload cannot tolerate the VM rebooting in order to attach networks" or "because my legacy workload requires the dynamic attachment of L2 networks" or "because my workload statefully tracks the attachment of networks" are some that I could think of.

Reply:
Thanks @dougbtv . 

I've added a few lines going over a subset of those. @phoracek is it clearer now ?

Reply:
Yeah, it looks great, thanks